test/functional/forall/atomic-ref/tests/test-forall-atomic-ref-math.hpp

```diff
@@ -220,73 +160,53 @@ void testAtomicRefCount(RAJA::RangeSegment seg,
 }
 
 
-template <typename ExecPolicy,
-         typename AtomicPolicy,
-         typename T,
-  RAJA::Index_type N>
-void testAtomicRefIntegral()
+TYPED_TEST_SUITE_P(SeqForallAtomicRefMathFunctionalTest);
+
+template <typename T>
+class SeqForallAtomicRefMathFunctionalTest : public ::testing::Test
 {
-  RAJA::RangeSegment seg(0, N);
+};
 
-  // initialize an array
 #if defined(RAJA_ENABLE_CUDA)
-  T *count = nullptr;
-  cudaErrchk(cudaMallocManaged((void **)&count, sizeof(T) * 1));
-  T *list;
-  cudaErrchk(cudaMallocManaged((void **)&list, sizeof(T) * N));
-  bool *hit;
-  cudaErrchk(cudaMallocManaged((void **)&hit, sizeof(bool) * N));
-  cudaErrchk(cudaDeviceSynchronize());
-#else
-  T *count  = new T[1];
-  T *list   = new T[N];
-  bool *hit = new bool[N];
-#endif
-
-  testAtomicRefCount<ExecPolicy, AtomicPolicy, T, PreIncCountOp  >(seg, count, list, hit);
-  testAtomicRefCount<ExecPolicy, AtomicPolicy, T, PostIncCountOp >(seg, count, list, hit);
-  testAtomicRefCount<ExecPolicy, AtomicPolicy, T, AddEqCountOp   >(seg, count, list, hit);
-  testAtomicRefCount<ExecPolicy, AtomicPolicy, T, FetchAddCountOp>(seg, count, list, hit);
+TYPED_TEST_SUITE_P(CudaForallAtomicRefMathFunctionalTest);
 
-  testAtomicRefCount<ExecPolicy, AtomicPolicy, T, PreDecCountOp  >(seg, count, list, hit);
-  testAtomicRefCount<ExecPolicy, AtomicPolicy, T, PostDecCountOp >(seg, count, list, hit);
-  testAtomicRefCount<ExecPolicy, AtomicPolicy, T, SubEqCountOp   >(seg, count, list, hit);
-  testAtomicRefCount<ExecPolicy, AtomicPolicy, T, FetchSubCountOp>(seg, count, list, hit);
-
-#if defined(RAJA_ENABLE_CUDA)
-  cudaErrchk(cudaFree(hit));
-  cudaErrchk(cudaFree(list));
-  cudaErrchk(cudaFree(count));
-#else
-  delete[] hit;
-  delete[] list;
-  delete[] count;
+template <typename T>
+class CudaForallAtomicRefMathFunctionalTest : public ::testing::Test
+{
+};
 #endif
-}
 
+#if defined(RAJA_ENABLE_HIP)
+TYPED_TEST_SUITE_P(HipForallAtomicRefMathFunctionalTest);
 
+template <typename T>
+class HipForallAtomicRefMathFunctionalTest : public ::testing::Test
+{
+};
+#endif
 
 template <typename ExecPolicy,
-         typename AtomicPolicy,
-         typename T,
-  RAJA::Index_type N>
-void testAtomicRefFloating()
+          typename AtomicPolicy,
+          typename WORKINGRES,
+          typename T>
+void testAtomicFunctionRefMath( RAJA::Index_type N )
 {
   RAJA::RangeSegment seg(0, N);
```

Comment:
We should be using Typed segments everywhere so we don't have to change this if we eventually remove these ty aliases

Reply:
Thanks, done.